library/firewall_lib.py

```diff
@@ -423,10 +525,66 @@ def main():
             )
         ):
             module.fail_json(
-                msg="The states present and absent can only be used in zone level operations (i.e. when no other parameters but zone and state are set)."
+                msg="present and absent only usable for zone or service operations"
```

Comment:
I think the more verbose error message is better in this case.  If you are running into issues with flake8 and black and line length, you can wrap like this:
```python
  msg=(
    "The states ......."
    "(i.e. when no ..."
  )
```